packages/date-time/src/components/Calendar/Calendar.base.tsx

```diff
@@ -0,0 +1,343 @@
+    firstWeekOfYear: FirstWeekOfYear.FirstDay,
+    dateTimeFormatter: defaultDateTimeFormatterCallbacks,
+    showSixWeeksByDefault: false,
+    workWeekDays: defaultWorkWeekDays,
+    showCloseButton: false,
+    allFocusable: false
+  };
+
+  private _dayPicker = React.createRef<ICalendarDay>();
+  private _monthPicker = React.createRef<ICalendarMonth>();
+
+  private _focusOnUpdate: boolean;
+
+  constructor(props: ICalendarProps) {
+    super(props);
+    const currentDate = props.value && !isNaN(props.value.getTime()) ? props.value : props.today || new Date();
+
+    this.state = {
+      selectedDate: currentDate,
+      navigatedDayDate: currentDate,
+      navigatedMonthDate: currentDate,
+
+      /** When showMonthPickerAsOverlay is active it overrides isMonthPickerVisible/isDayPickerVisible props
+       (These props permanently set the visibility of their respective calendars). */
+      isMonthPickerVisible: this.props.showMonthPickerAsOverlay ? false : this.props.isMonthPickerVisible,
+      isDayPickerVisible: this.props.showMonthPickerAsOverlay ? true : this.props.isDayPickerVisible
+    };
+
+    this._focusOnUpdate = false;
+  }
+
+  public componentWillReceiveProps(nextProps: ICalendarProps): void {
+    const { value, today = new Date() } = nextProps;
+
+    this.setState({
+      selectedDate: value || today,
+      navigatedDayDate: value || today,
+      navigatedMonthDate: value || today
+    });
+  }
+
+  public componentDidUpdate(): void {
+    if (this._focusOnUpdate) {
+      this.focus();
+      this._focusOnUpdate = false;
+    }
+  }
+
+  public render(): JSX.Element {
+    const rootClass = 'ms-DatePicker';
+    const {
+      firstDayOfWeek,
+      dateRangeType,
+      strings,
+      showMonthPickerAsOverlay,
+      showGoToToday,
+      highlightCurrentMonth,
+      highlightSelectedMonth,
+      navigationIcons,
+      minDate,
+      maxDate,
+      className,
+      showCloseButton,
+      allFocusable,
+      styles,
+      showWeekNumbers,
+      theme
+    } = this.props;
+    const { selectedDate, navigatedDayDate, navigatedMonthDate, isMonthPickerVisible, isDayPickerVisible } = this.state;
+    const onHeaderSelect = showMonthPickerAsOverlay ? this._onHeaderSelect : undefined;
+    const monthPickerOnly = !showMonthPickerAsOverlay && !isDayPickerVisible;
+    const overlayedWithButton = showMonthPickerAsOverlay && showGoToToday;
+
+    const classes = getClassNames(styles, {
+      theme: theme!,
+      className,
+      isMonthPickerVisible: isMonthPickerVisible,
+      isDayPickerVisible: isDayPickerVisible,
+      monthPickerOnly: monthPickerOnly,
+      showMonthPickerAsOverlay: showMonthPickerAsOverlay,
+      overlayedWithButton: overlayedWithButton,
+      showGoToToday: showGoToToday,
+      showWeekNumbers: showWeekNumbers
+    });
+
+    return (
+      <div
+        className={css(rootClass, classes.root, className, 'ms-slideDownIn10')}
+        role="application"
```

Comment:
>application [](start = 14, length = 11)

Do we really want application here?  The Calendar is composed of standard HTML components, right?

Reply:
Based on this article you found: https://www.marcozehe.de/2012/02/06/if-you-use-the-wai-aria-role-application-please-do-so-wisely/
no, we don't actually need that role. Removing.